src/libraries/System.Diagnostics.EventLog/tests/EventLogMessagesTests.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.Eventing.Reader;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+using Xunit;
+
+namespace System.Diagnostics.Tests
+{
+    public class EventLogMessagesTests
+    {
+        [Fact]
+        public void EventLogMessagesContainsNoTypes()
+        {
+            Assembly messageAssembly = Assembly.Load("System.Diagnostics.EventLog.Messages");
+            Assert.NotNull(messageAssembly);
+            Assert.Empty(messageAssembly.GetTypes());
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
+        [InlineData(65535)]
+        public void CanFormatMessage(uint messageId)
+        {
+            string messageDllPath = Path.Combine(Path.GetDirectoryName(typeof(EventLog).Assembly.Location), "System.Diagnostics.EventLog.Messages.dll");
+            Assert.True(File.Exists(messageDllPath));
+            using SafeLibraryHandle hMessageDll = Interop.Kernel32.LoadLibraryExW(messageDllPath, IntPtr.Zero, Interop.Kernel32.LOAD_LIBRARY_AS_DATAFILE);
+
+            string messageString = "hello message";
+            char[] buffer = new char[1024];
+            GCHandle pinnedString = GCHandle.Alloc(messageString, GCHandleType.Pinned);
+            try
+            {
+                IntPtr[] insertion = new[] { pinnedString.AddrOfPinnedObject() };
```

Comment:
How does this differ from just doing `fixed (char* p = messageString)` ? Although, that would give a pointer to a char array rather than an IntPtr array, won't the value be the same?

Reply:
I just copied the product code that was using the same API.  I'll see if I can simplify the test.